client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-assign-trial-plan/index.tsx

```diff
@@ -26,15 +27,16 @@ const SiteMigrationAssignTrialPlanStep: Step = ( { navigation } ) => {
 	} );
 
 	useEffect( () => {
-		if ( site ) {
+		if ( site && ! isAddingTrial ) {
+			setIsAddingTrial( true );
 			const assignMigrationTrialPlan = () => {
 				site.ID && addHostingTrial( site.ID, PLAN_MIGRATION_TRIAL_MONTHLY, HOSTING_INTENT_MIGRATE );
 			};
 
 			assignMigrationTrialPlan();
 		}
 		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
probably this comment should be removed, since now we are checking the deps.

Reply:
I left the comment in place because we're still not checking for the `assignMigrationTrialPlan` dependency. I assumed there was a reason for not having `assignMigrationTrialPlan` as a dependency, but maybe there isn't a reason?

Reply:
wait, isn't `assignMigrationTrialPlan` defined inside the `useEffect`? probably I am misunderstanding something 🤔

Reply:
Sorry, I misspoke! I meant `addHostingTrial` isn't part of the dependencies. But I think you're right and I should add it and remove the comment.

Reply:
Done!